desktop/apps/article2/__tests__/routes.test.js

```diff
@@ -26,21 +29,157 @@ describe('#index', () => {
+      const data = {
+        article: _.extend({}, fixtures.article, {
+          slug: 'foobar',
+          channel_id: '123'
+        })
+      }
+      RoutesRewireApi.__Rewire__(
+        'positronql',
+        sinon.stub().returns(Promise.resolve(data))
+      )
+      const renderReactLayout = sinon.stub()
+      RoutesRewireApi.__Rewire__('renderReactLayout', renderReactLayout)
+      RoutesRewireApi.__Rewire__('sd', {ARTSY_EDITORIAL_CHANNEL: '123'})
+      index(req, res, next)
+        .then(() => {
+          renderReactLayout.args[0][0].data.article.title.should.equal('Top Ten Booths')
+          renderReactLayout.args[0][0].locals.assetPackage.should.equal('article2')
+          done()
+        })
+    })
+
+    it('redirects to the correct slug if it does not match', (done) => {
+      const data = {
+        article: _.extend({}, fixtures.article, {
+          slug: 'zoobar',
+          channel_id: '123'
+        })
+      }
+      RoutesRewireApi.__Rewire__(
+        'positronql',
+        sinon.stub().returns(Promise.resolve(data))
+      )
+      RoutesRewireApi.__Rewire__('sd', {ARTSY_EDITORIAL_CHANNEL: '123'})
+      index(req, res, next)
+        .then(() => {
+          res.redirect.args[0][0].should.equal('/article2/zoobar')
+          done()
+        })
+    })
+
+    it('renders classic mode if article is not editorial', (done) => {
+      const data = {
+        article: new Article(_.extend({}, fixtures.article, {
+          slug: 'foobar',
+          channel_id: '456'
+        })),
+        channel: new Channel()
+      }
+      RoutesRewireApi.__Rewire__(
+        'positronql',
+        sinon.stub().returns(Promise.resolve(data))
+      )
+      RoutesRewireApi.__Rewire__('sd', {ARTSY_EDITORIAL_CHANNEL: '123'})
+      Article.prototype.fetchWithRelated = sinon.stub().yieldsTo('success', data)
+      RoutesRewireApi.__Rewire__('Article', Article)
+      index(req, res, next)
+        .then(() => {
+          res.render.args[0][0].should.equal('article')
+          done()
+        })
+    })
+  })
+
+  describe('#amp', () => {
+    it('renders amp page', (done) => {
+      const data = {
+        article: new Article(_.extend({}, fixtures.article, {
+          slug: 'foobar',
+          channel_id: '456',
+          sections: [],
+          featured: true,
+          published: true
+        })),
+        channel: new Channel()
+      }
+      RoutesRewireApi.__Rewire__(
+        'positronql',
+        sinon.stub().returns(Promise.resolve(data))
+      )
+      RoutesRewireApi.__Rewire__('sd', {ARTSY_EDITORIAL_CHANNEL: '123'})
+      Article.prototype.fetchWithRelated = sinon.stub().yieldsTo('success', data)
+      RoutesRewireApi.__Rewire__('Article', Article)
+      amp(req, res, next)
+      res.render.args[0][0].should.equal('amp_article')
+      done()
+    })
+
+    it('skips if it has an artwork or image section', (done) => {
+      const data = {
```

Comment:
Yeah naming could be better -- basically we can't render articles that have `artwork` or `image` sections in AMP because they don't have image height + width dimensions saved. Image dimensions are a requirement for rendering images in AMP. See https://github.com/artsy/force/blob/master/desktop/models/article.coffee#L241